### PR TITLE
api for npl suggestions

### PR DESCRIPTION
**Pull request by yneves**
Still working on client side, but you can start reviewing API when you have a chance.

### CODE REVIEW

File: src/db/_documents.js [hunk 1]
```diff
@@ -72,6 +72,21 @@ function * createIfNeeded(doc) {
...
+function * searchNplDocuments(firmName, clientNumber, nonPatentBib) {
+  if (!firmName || !clientNumber || !nonPatentBib) {
+    return {};
+  }
+
+  const query = _getBaseQuery(firmName, clientNumber)
+    .where('type=?', 'Other')
+    .where('nonPatentBib LIKE ?', `%${nonPatentBib}%`)
+    .field('documentNumber')
+    .field('nonPatentBib');
```
Comment (ntgn81) [inline src/db/_documents.js]: @yneves , try out MATCH instead. That's what the old UI was using.

https://dev.mysql.com/doc/refman/5.7/en/fulltext-search.html#function_match

**Comment by yneves** [comment]
@ntgn81 It's using MATCH now and also decoding the text, missed that at first.

File: src/api/documents.js [hunk 2]
```diff
@@ -20,9 +20,16 @@ router
   .get('/update-cited', updateCited)
   .get('/:id/replace-doc-number/:documentNumber', replaceDocNumber)
   .get('/retrieve/:countryCode/:number', retrieve)
+  .get('/search-npl', searchNpl)
   .put('/:id', update)
   .put('/:id/upload-file/:mode', uploadFile);
 
+function * searchNpl() {
+  const {firmName, clientNumber, nonPatentBib} = this.query;
```
Comment (ntgn81) [inline src/api/documents.js]: We need to update node version in package json to 6.x for destructuring to work

File: src/api/documents.js [hunk 3]
```diff
@@ -20,9 +20,16 @@ router
...
+  const suggestions = yield db.documents.searchNplDocuments(firmName, clientNumber, nonPatentBib);
+  this.body = _.merge({}, suggestions);
```
Comment (ntgn81) [inline src/api/documents.js]: Unnecessary merge, I think

**Comment by yneves** [comment]
Done both

**Pull request merged**